core/compatible.py

```diff
@@ -71,15 +71,15 @@ def check_for_requirements(start_api_server):
     api_config = api_configuration()
     connection_timeout = api_config["api_database_connection_timeout"]
     try:
-        import pymongo
-        import netaddr
-        import flask
-        del netaddr
-        del flask
+        with open(os.path.join(os.getcwd(), 'requirements.txt'), 'r') as requirements:
+            for dependency in requirements:
+                dependency_name = dependency.split('=')[0]
```

Comment:
i'm not sure if all packages which have package name are imported as same?
because if you install tensorflow-gpu for instance, still you would import tensorflow and not tensorflow-gpu

Reply:
yes that's possible

Reply:
@dhirensr what can be the possible workaround

Reply:
i think this code is better than what we have today; what do you think?

Reply:
@dhirensr

Reply:
yes it's better, but i'm not sure if it will work for all packages.

Reply:
the approach is good ,but it will unnecessary import all the requirements, some of which are not even needed(sounds somewhat inefficient to me)

Reply:
```python
        import pkg_resources
        installed_packages = pkg_resources.working_set
        installed_packages_list = sorted(["%s==%s" % (i.key, i.version)
                                          for i in installed_packages])
        with open(os.path.join(os.getcwd(), 'requirements.txt'), 'r') as requirements:
            for dependency in requirements:
                if dependency.strip() not in installed_packages_list:
                    raise Exception("Package not installed")
```
@Ali-Razmjoo @dhirensr will this approach work